src/socket.rs

```diff
@@ -1391,6 +1397,10 @@ mod test {
         ($e:expr) => (match $e { Ok(e) => e, Err(e) => panic!("{:?}", e) })
     }
 
+    macro_rules! mutetry {
```

Comment:
this is only being used in the test and the only purpose is just to allow the tests can continue with the 100 iterations when there is connection failure happens.
i.e. the previous functionality test is now being expanded as a measurement test, which the previous error handling policy of `panic` is not becomeing `logging`.